code/Model/EditableFormField/EditableFileField.php

```diff
@@ -60,14 +64,21 @@ public function getCMSFields()
             )
         );
 
-        $fields->addFieldToTab("Root.Main", LiteralField::create(
-            'FileUploadWarning',
-            '<p class="alert alert-info">' . _t(
-                'SilverStripe\\UserForms\\Model\\UserDefinedForm.FileUploadWarning',
-                'Files uploaded through this field could be publicly accessible if the exact URL is known'
-            )
-            . '</p>'
-        ), 'Type');
+        $folderIsRestricted = Member::actAs(null, function () {
+            return $this->FolderID && !$this->Folder()->canView();
+        });
+
+        if (!$folderIsRestricted) {
+            $fields->addFieldToTab("Root.Main", LiteralField::create(
```

Comment:
I think we should avoid including HTML in our PHP code. If practical please move this to a template.

Reply:
Using a `LiteralField` with inline HTML is, for better or worse, a standard development pattern for notices at the moment. We should look into producing a dedicated `NoticeField` in the future, but for now I think this is simple and common enough that it should be okay to proceed with.